testDDRExtJunit_MonitorsAndDeadlock '!monitors  deadlock' output is null or empty

https://ci.eclipse.org/openj9/job/Test_openjdk16_j9_extended.functional_ppc64_aix_Nightly_testList_0/86
testDDRExtJunit_MonitorsAndDeadlock6_1
```
     [exec] JVMDUMP039I Processing dump event "throw", detail "j9vm/test/corehelper/HelperExceptionForCoreGeneration" at 2021/05/09 23:22:49 - please wait.
     [exec] JVMDUMP032I JVM requested System dump using '/home/jenkins/workspace/Test_openjdk16_j9_extended.functional_ppc64_aix_Nightly_testList_0/openjdk-tests/TKG/../TKG/output_16206157323654/testDDRExtJunit_MonitorsAndDeadlock6_1/DDREXT.J9CORE.DMP' in response to an event
     [exec] HelperExceptionForCoreGeneration is thrown and caught successfuly.
     [exec] Forcefully terminating process.
     [exec] addTestJitHint adding hint to openj9/internal/tools/attach/target/AttachHandler$1.<clinit>
     [exec] JVMDUMP010I System dump written to /home/jenkins/workspace/Test_openjdk16_j9_extended.functional_ppc64_aix_Nightly_testList_0/openjdk-tests/TKG/../TKG/output_16206157323654/testDDRExtJunit_MonitorsAndDeadlock6_1/DDREXT.J9CORE.DMP
     [exec] JVMDUMP013I Processed dump event "throw", detail "j9vm/test/corehelper/HelperExceptionForCoreGeneration".
     [exec] addTestJitHint adding hint to java/util/Vector$Itr.<init>
     [exec] addTestJitHint adding hint to java/lang/Thread$State.values

TCK.destroy.cache:
     [echo] Destroying cache
     [echo] Running j9vm.test.corehelper.StackMapCoreGenerator to destroy the cache
     [echo] Using JVM : /home/jenkins/workspace/Test_openjdk16_j9_extended.functional_ppc64_aix_Nightly_testList_0/openjdkbinary/j2sdk-image/bin/java
     [echo] classname = "j9vm.test.corehelper.StackMapCoreGenerator"
     [echo] Java VM Args:
     [echo]   jvmarg = -Xshareclasses:name=ddrextjunitSCC,destroy
     [java] JVMSHRC806I Compressed references persistent shared cache "ddrextjunitSCC" has been destroyed. Use option -Xnocompressedrefs if you want to destroy a non-compressed references cache.
     [java] Java Result: 1

TCK.run.tests.ddrext:
     [echo] Running the DDR Extension Test
     [java] [SetupConfig] [INFO] Created new DDR Interactive instance using core file : /home/jenkins/workspace/Test_openjdk16_j9_extended.functional_ppc64_aix_Nightly_testList_0/openjdk-tests/TKG/../TKG/output_16206157323654/testDDRExtJunit_MonitorsAndDeadlock6_1/DDREXT.J9CORE.DMP
     [java] [AutoRun] [INFO] Running 1 tests for /home/jenkins/workspace/Test_openjdk16_j9_extended.functional_ppc64_aix_Nightly_testList_0/openjdk-tests/TKG/../TKG/output_16206157323654/testDDRExtJunit_MonitorsAndDeadlock6_1/DDREXT.J9CORE.DMP
     [java] [AutoRun] [INFO] Starting testDeadlock6(j9vm.test.ddrext.junit.deadlock.TestDeadlockCase6)
     [java] [DDRExtTesterBase] [INFO] Validation started for : '!monitors  deadlock'
     [java] [AutoRun] [INFO] Finished testDeadlock6(j9vm.test.ddrext.junit.deadlock.TestDeadlockCase6)
     [java] 
     [java] [AutoRun] [INFO] ================Test Result for /home/jenkins/workspace/Test_openjdk16_j9_extended.functional_ppc64_aix_Nightly_testList_0/openjdk-tests/TKG/../TKG/output_16206157323654/testDDRExtJunit_MonitorsAndDeadlock6_1/DDREXT.J9CORE.DMP==================
     [java] [AutoRun] [INFO] Errors:   0 out of 1 test cases.
     [java] [AutoRun] [INFO] Failures: 1 out of 1 test cases.
     [java] [AutoRun] [INFO] Total Error/Failures: 1
     [java] [DDRExtTesterBase] [ERROR] '!monitors  deadlock' output is null or empty
     [java] junit.framework.AssertionFailedError
     [java] 	at junit.framework.Assert.fail(Assert.java:48)
     [java] 	at junit.framework.Assert.assertTrue(Assert.java:20)
     [java] 	at junit.framework.Assert.assertTrue(Assert.java:27)
     [java] 	at j9vm.test.ddrext.junit.deadlock.TestDeadlockCase6.testDeadlock6(TestDeadlockCase6.java:47)
     [java] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
     [java] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
     [java] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
     [java] 	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
     [java] 	at junit.framework.TestCase.runTest(TestCase.java:168)
     [java] 	at junit.framework.TestCase.runBare(TestCase.java:134)
     [java] 	at junit.framework.TestResult$1.protect(TestResult.java:110)
     [java] 	at junit.framework.TestResult.runProtected(TestResult.java:128)
     [java] 	at junit.framework.TestResult.run(TestResult.java:113)
     [java] 	at junit.framework.TestCase.run(TestCase.java:124)
     [java] 	at junit.framework.TestSuite.runTest(TestSuite.java:243)
     [java] 	at j9vm.test.ddrext.junit.DDRTestSuite.run(DDRTestSuite.java:74)
     [java] 	at j9vm.test.ddrext.junit.DDRTestSuite.run(DDRTestSuite.java:69)
     [java] 	at j9vm.test.ddrext.junit.DDRTestSuite.run(DDRTestSuite.java:55)
     [java] 	at j9vm.test.ddrext.AutoRun.runTest(AutoRun.java:275)
     [java] 	at j9vm.test.ddrext.AutoRun.main(AutoRun.java:110)
     [java] [AutoRun] [ERROR] testDeadlock6(j9vm.test.ddrext.junit.deadlock.TestDeadlockCase6)
```

https://ci.eclipse.org/openj9/job/Test_openjdk11_j9_extended.functional_x86-64_windows_Nightly_testList_1/306
```
     [exec] addTestJitHint class java/lang/Byte$ByteCache not in the cache
     [exec] addTestJitHint class java/lang/invoke/Insert2Handle not in the cache
     [exec] Waiting...
     [exec] spawnFirstThreadtest6 :: run()
     [exec] spawnSecondThreadtest6 :: run()
     [exec] spawnSecondThreadtest6 :: block indefinitely on lock1test6
     [exec] JVMDUMP039I Processing dump event "throw", detail "j9vm/test/corehelper/HelperExceptionForCoreGeneration" at 2021/05/15 23:34:11 - please wait.
     [exec] JVMDUMP032I JVM requested System dump using 'F:/Users/jenkins/workspace/Test_openjdk11_j9_extended.functional_x86-64_windows_Nightly_testList_1/openjdk-tests/\TKG\output_16211376163392\testDDRExtJunit_MonitorsAndDeadlock6_0/DDREXT.J9CORE.DMP' in response to an event
     [exec] JVMDUMP010I System dump written to F:/Users/jenkins/workspace/Test_openjdk11_j9_extended.functional_x86-64_windows_Nightly_testList_1/openjdk-tests/\TKG\output_16211376163392\testDDRExtJunit_MonitorsAndDeadlock6_0/DDREXT.J9CORE.DMP
     [exec] JVMDUMP013I Processed dump event "throw", detail "j9vm/test/corehelper/HelperExceptionForCoreGeneration".
     [exec] HelperExceptionForCoreGeneration is thrown and caught successfuly.
     [exec] Forcefully terminating process.
     [exec] addTestJitHint class java/util/IdentityHashMap$KeyIterator not in the cache
     [exec] addTestJitHint class java/util/IdentityHashMap$IdentityHashMapIterator not in the cache
     [exec] addTestJitHint class openj9/internal/tools/attach/target/AttachHandler$1 not in the cache
     [exec] addTestJitHint class java/util/Vector$Itr not in the cache
     [exec] addTestJitHint class java/lang/Thread$State not in the cache

TCK.destroy.cache:
     [echo] Destroying cache
     [echo] Running j9vm.test.corehelper.StackMapCoreGenerator to destroy the cache
     [echo] Using JVM : F:/Users/jenkins/workspace/Test_openjdk11_j9_extended.functional_x86-64_windows_Nightly_testList_1/openjdkbinary/j2sdk-image\bin\java
     [echo] classname = "j9vm.test.corehelper.StackMapCoreGenerator"
     [echo] Java VM Args:
     [echo]   jvmarg = -Xshareclasses:name=ddrextjunitSCC,destroy
     [java] JVMSHRC806I Compressed references persistent shared cache "ddrextjunitSCC" has been destroyed. Use option -Xnocompressedrefs if you want to destroy a non-compressed references cache.
     [java] Java Result: 1

TCK.run.tests.ddrext:
     [echo] Running the DDR Extension Test
     [java] [SetupConfig] [INFO] Created new DDR Interactive instance using core file : F:\Users\jenkins\workspace\Test_openjdk11_j9_extended.functional_x86-64_windows_Nightly_testList_1\openjdk-tests\TKG\output_16211376163392\testDDRExtJunit_MonitorsAndDeadlock6_0\DDREXT.J9CORE.DMP
     [java] [AutoRun] [INFO] Running 1 tests for F:\Users\jenkins\workspace\Test_openjdk11_j9_extended.functional_x86-64_windows_Nightly_testList_1\openjdk-tests\TKG\output_16211376163392\testDDRExtJunit_MonitorsAndDeadlock6_0\DDREXT.J9CORE.DMP
     [java] [AutoRun] [INFO] Starting testDeadlock6(j9vm.test.ddrext.junit.deadlock.TestDeadlockCase6)
     [java] [DDRExtTesterBase] [ERROR] '!monitors  deadlock' output is null or empty
     [java] [DDRExtTesterBase] [INFO] Validation started for : '!monitors  deadlock'
     [java] junit.framework.AssertionFailedError
     [java] [AutoRun] [INFO] Finished testDeadlock6(j9vm.test.ddrext.junit.deadlock.TestDeadlockCase6)
     [java] 	at junit.framework.Assert.fail(Assert.java:48)
     [java] 	at junit.framework.Assert.assertTrue(Assert.java:20)
     [java] 
     [java] 	at junit.framework.Assert.assertTrue(Assert.java:27)
     [java] 	at j9vm.test.ddrext.junit.deadlock.TestDeadlockCase6.testDeadlock6(TestDeadlockCase6.java:47)
     [java] [AutoRun] [INFO] ================Test Result for F:\Users\jenkins\workspace\Test_openjdk11_j9_extended.functional_x86-64_windows_Nightly_testList_1\openjdk-tests\TKG\output_16211376163392\testDDRExtJunit_MonitorsAndDeadlock6_0\DDREXT.J9CORE.DMP==================
     [java] [AutoRun] [INFO] Errors:   0 out of 1 test cases.
     [java] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
     [java] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
     [java] [AutoRun] [INFO] Failures: 1 out of 1 test cases.
     [java] [AutoRun] [INFO] Total Error/Failures: 1
     [java] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
     [java] 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
     [java] 	at junit.framework.TestCase.runTest(TestCase.java:168)
     [java] 	at junit.framework.TestCase.runBare(TestCase.java:134)
     [java] 	at junit.framework.TestResult$1.protect(TestResult.java:110)
     [java] 	at junit.framework.TestResult.runProtected(TestResult.java:128)
     [java] 	at junit.framework.TestResult.run(TestResult.java:113)
     [java] 	at junit.framework.TestCase.run(TestCase.java:124)
     [java] 	at junit.framework.TestSuite.runTest(TestSuite.java:243)
     [java] 	at j9vm.test.ddrext.junit.DDRTestSuite.run(DDRTestSuite.java:74)
     [java] 	at j9vm.test.ddrext.junit.DDRTestSuite.run(DDRTestSuite.java:69)
     [java] 	at j9vm.test.ddrext.junit.DDRTestSuite.run(DDRTestSuite.java:55)
     [java] 	at j9vm.test.ddrext.AutoRun.runTest(AutoRun.java:275)
     [java] 	at j9vm.test.ddrext.AutoRun.main(AutoRun.java:110)
     [java] [AutoRun] [ERROR] testDeadlock6(j9vm.test.ddrext.junit.deadlock.TestDeadlockCase6)
```

https://openj9-jenkins.osuosl.org/job/Test_openjdk8_j9_extended.functional_x86-64_linux_Nightly/27
```
TCK.run.tests.ddrext:
     [echo] Running the DDR Extension Test
     [java] openjdk version "1.8.0_302-internal"
     [java] OpenJDK Runtime Environment (build 1.8.0_302-internal-_2021_06_21_01_06-b00)
     [java] Eclipse OpenJ9 VM (build master-d0a0f90, JRE 1.8.0 Linux amd64-64-Bit Compressed References 20210621_34 (JIT enabled, AOT enabled)
     [java] OpenJ9   - d0a0f90
     [java] OMR      - b7eb963
     [java] JCL      - be81923 based on jdk8u302-b05)
     [java] 
     [java] [SetupConfig] [INFO] Created new DDR Interactive instance using core file : /home/jenkins/workspace/Test_openjdk8_j9_extended.functional_x86-64_linux_Nightly_testList_0/aqa-tests/TKG/../TKG/output_16242513202013/testDDRExtJunit_MonitorsAndDeadlock6_0/DDREXT.J9CORE.DMP
     [java] [AutoRun] [INFO] Running 1 tests for /home/jenkins/workspace/Test_openjdk8_j9_extended.functional_x86-64_linux_Nightly_testList_0/aqa-tests/TKG/../TKG/output_16242513202013/testDDRExtJunit_MonitorsAndDeadlock6_0/DDREXT.J9CORE.DMP
     [java] [AutoRun] [INFO] Starting testDeadlock6(j9vm.test.ddrext.junit.deadlock.TestDeadlockCase6)
     [java] [DDRExtTesterBase] [INFO] Validation started for : '!monitors  deadlock'
     [java] [AutoRun] [INFO] Finished testDeadlock6(j9vm.test.ddrext.junit.deadlock.TestDeadlockCase6)
     [java] 
     [java] [AutoRun] [INFO] ================Test Result for /home/jenkins/workspace/Test_openjdk8_j9_extended.functional_x86-64_linux_Nightly_testList_0/aqa-tests/TKG/../TKG/output_16242513202013/testDDRExtJunit_MonitorsAndDeadlock6_0/DDREXT.J9CORE.DMP==================
     [java] [AutoRun] [INFO] Errors:   0 out of 1 test cases.
     [java] [AutoRun] [INFO] Failures: 1 out of 1 test cases.
     [java] [AutoRun] [INFO] Total Error/Failures: 1
     [java] [DDRExtTesterBase] [ERROR] '!monitors  deadlock' output is null or empty
     [java] junit.framework.AssertionFailedError
     [java] 	at junit.framework.Assert.fail(Assert.java:48)
     [java] 	at junit.framework.Assert.assertTrue(Assert.java:20)
     [java] 	at junit.framework.Assert.assertTrue(Assert.java:27)
     [java] 	at j9vm.test.ddrext.junit.deadlock.TestDeadlockCase6.testDeadlock6(TestDeadlockCase6.java:47)
     [java] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
     [java] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
     [java] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
     [java] 	at java.lang.reflect.Method.invoke(Method.java:498)
     [java] 	at junit.framework.TestCase.runTest(TestCase.java:168)
     [java] 	at junit.framework.TestCase.runBare(TestCase.java:134)
     [java] 	at junit.framework.TestResult$1.protect(TestResult.java:110)
     [java] 	at junit.framework.TestResult.runProtected(TestResult.java:128)
     [java] 	at junit.framework.TestResult.run(TestResult.java:113)
     [java] 	at junit.framework.TestCase.run(TestCase.java:124)
     [java] 	at junit.framework.TestSuite.runTest(TestSuite.java:243)
     [java] 	at j9vm.test.ddrext.junit.DDRTestSuite.run(DDRTestSuite.java:74)
     [java] 	at j9vm.test.ddrext.junit.DDRTestSuite.run(DDRTestSuite.java:69)
     [java] 	at j9vm.test.ddrext.junit.DDRTestSuite.run(DDRTestSuite.java:55)
     [java] 	at j9vm.test.ddrext.AutoRun.runTest(AutoRun.java:275)
     [java] 	at j9vm.test.ddrext.AutoRun.main(AutoRun.java:110)
     [java] [AutoRun] [ERROR] testDeadlock6(j9vm.test.ddrext.junit.deadlock.TestDeadlockCase6)
```

https://openj9-jenkins.osuosl.org/job/Test_openjdk8_j9_extended.functional_x86-64_mac_Nightly_testList_0/297
testDDRExtJunit_MonitorsAndDeadlock6_0 `-Xjit -Xgcpolicy:optthruput -Xnocompressedrefs`

https://openj9-artifactory.osuosl.org/artifactory/ci-openj9/Test/Test_openjdk8_j9_extended.functional_x86-64_mac_Nightly_testList_0/297/functional_test_output.tar.gz

```
     [java] [AutoRun] [INFO] ================Test Result for /Users/jenkins/workspace/Test_openjdk8_j9_extended.functional_x86-64_mac_Nightly_testList_0/aqa-tests/TKG/../TKG/output_16552601287434/testDDRExtJunit_MonitorsAndDeadlock6_0/DDREXT.J9CORE.DMP==================
     [java] [AutoRun] [INFO] Errors:   0 out of 1 test cases.
     [java] [AutoRun] [INFO] Failures: 1 out of 1 test cases.
     [java] [AutoRun] [INFO] Total Error/Failures: 1
     [java] [DDRExtTesterBase] [ERROR] '!monitors  deadlock' output is null or empty
     [java] junit.framework.AssertionFailedError
     [java] 	at junit.framework.Assert.fail(Assert.java:48)
     [java] 	at junit.framework.Assert.assertTrue(Assert.java:20)
     [java] 	at junit.framework.Assert.assertTrue(Assert.java:27)
     [java] 	at j9vm.test.ddrext.junit.deadlock.TestDeadlockCase6.testDeadlock6(TestDeadlockCase6.java:47)
```

https://openj9-jenkins.osuosl.org/job/Test_openjdk11_j9_extended.functional_aarch64_mac_Nightly_testList_1/80
testDDRExtJunit_MonitorsAndDeadlock6_0

https://openj9-artifactory.osuosl.org/artifactory/ci-openj9/Test/Test_openjdk11_j9_extended.functional_aarch64_mac_Nightly_testList_1/80/functional_test_output.tar.gz

```
     [java] [AutoRun] [INFO] ================Test Result for /Users/jenkins/workspace/Test_openjdk11_j9_extended.functional_aarch64_mac_Nightly_testList_1/aqa-tests/TKG/../TKG/output_16554339382269/testDDRExtJunit_MonitorsAndDeadlock6_0/DDREXT.J9CORE.DMP==================
     [java] [AutoRun] [INFO] Errors:   0 out of 1 test cases.
     [java] [AutoRun] [INFO] Failures: 1 out of 1 test cases.
     [java] [AutoRun] [INFO] Total Error/Failures: 1
     [java] [DDRExtTesterBase] [ERROR] '!monitors  deadlock' output is null or empty
     [java] junit.framework.AssertionFailedError
     [java] 	at junit.framework.Assert.fail(Assert.java:48)
     [java] 	at junit.framework.Assert.assertTrue(Assert.java:20)
     [java] 	at junit.framework.Assert.assertTrue(Assert.java:27)
     [java] 	at j9vm.test.ddrext.junit.deadlock.TestDeadlockCase6.testDeadlock6(TestDeadlockCase6.java:47)
```

--- a/test/functional/DDR_Test/src/j9vm/test/corehelper/DeadlockCoreGenerator.java
+++ b/test/functional/DDR_Test/src/j9vm/test/corehelper/DeadlockCoreGenerator.java
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2001, 2018 IBM Corp. and others
+ * Copyright (c) 2001, 2022 IBM Corp. and others
  *
  * This program and the accompanying materials are made available under
  * the terms of the Eclipse Public License 2.0 which accompanies this
@@ -23,8 +23,26 @@ package j9vm.test.corehelper;
 
 import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.TimeUnit;
 
 public final class DeadlockCoreGenerator {
+
+	private static final class LockObject {
+
+		final String name;
+
+		LockObject(String name) {
+			super();
+			this.name = name;
+		}
+
+		@Override
+		public String toString() {
+			return "lock: " + name;
+		}
+
+	}
 
 	/*
 	 * **********
@@ -39,13 +57,11 @@ public final class DeadlockCoreGenerator {
 	 *
 	 * N.B. A native thread can never hold/want an object monitor,
 	 * because it must be attached first (i.e. it's a J9VMThread / Java thread).
-	 *
-	 */	
-	
-	public static final long WAIT_TIME_MILIS = 5000; // 5 seconds.
-	
-	static 
-	{
+	 */
+
+	public static final long WAIT_TIME_MILIS = TimeUnit.SECONDS.toMillis(5);
+
+	static {
 		try {
 			System.loadLibrary("j9ben");
 		} catch (UnsatisfiedLinkError e) {
@@ -55,37 +71,33 @@ public final class DeadlockCoreGenerator {
 			System.exit(1);
 		}
 	}
-	
-	public static void main(final String[] args) throws InterruptedException, NoSuchMethodException, 
-		SecurityException, IllegalAccessException, IllegalArgumentException, InvocationTargetException {
-		
+
+	public static void main(final String[] args) throws Exception {
 		System.out.println("Running deadlock tester...");
-		
-		final Method testCase = DeadlockCoreGenerator.class.getDeclaredMethod("testCase"  + args[0]);
-		
+
+		String testName = "testCase" + args[0];
+		final Method testCase = DeadlockCoreGenerator.class.getDeclaredMethod(testName);
+
 		Thread testThread = new Thread() {
-			
+			@Override
 			public void run() {
 				try {
-					
-					testCase.invoke(null, new Object[] {});
-					
-				} catch (IllegalAccessException e) {
+					testCase.invoke(null);
+				} catch (IllegalAccessException | IllegalArgumentException | InvocationTargetException e) {
 					e.printStackTrace();
-				} catch (IllegalArgumentException e) {
-					e.printStackTrace();
-				} catch (InvocationTargetException e) {
-					e.printStackTrace();
-				}	
+				}
 			}
 		};
-		
+
 		testThread.setName("Java Deadlock Test Thread");
 		testThread.start();
-				
+
+		// Wait for positive indication that threads are about to deadlock.
+		waitForThreadsToApproachDeadlock(testName);
+
 		System.out.println("Waiting...");
 		Thread.sleep(WAIT_TIME_MILIS); // This is a guesstimate, but should be long enough to reach the desired state.
-		
+
 		try {
 			throw new HelperExceptionForCoreGeneration();
 		} catch (HelperExceptionForCoreGeneration e) {
@@ -94,171 +106,200 @@ public final class DeadlockCoreGenerator {
 			// In all but the native-only deadlock, the JVM won't quit since not all threads died.
 			System.exit(0); // Note that this will only run once the dump is done (exclusive!).
 		}
-		
 	}
-	
+
 	/* *****************************
 	 * Utility Methods for all Tests
 	 * *****************************/
-	
+
 	public static native void setup();
 	public static native void enterFirstMonitor();
 	public static native void enterSecondMonitor();
 	public static native void spawnNativeThread();
 	public static native void createNativeDeadlock();
-	
-	private static final Object mutex = new Object();
-	
+
+	private static final Object mutex = new LockObject("mutex");
+
+	private static final Semaphore deadlock = new Semaphore(0);
+
+	private static void aboutToDeadlock() {
+		deadlock.release();
+	}
+
+	private static void waitForThreadsToApproachDeadlock(String testName) throws InterruptedException {
+		int permits;
+
+		switch (testName) {
+		case "testCase4":
+			permits = 1;
+			break;
+		case "testCase5":
+			permits = 0;
+			break;
+		default:
+			permits = 2;
+			break;
+		}
+
+		if (permits != 0) {
+			System.out.println("Waiting for threads to approach deadlock...");
+			deadlock.acquire(permits);
+		}
+	}
+
 	/* *****************************
 	 *         Test Case #1
-	 *	     JOO, JOO deadlock.
+	 *       JOO, JOO deadlock.
 	 * *****************************/
-	
-	public static void testCase1() throws InterruptedException 
-	{
-		final Object firstLock = new Object();
-		final Object secondLock = new Object();
-		
-		synchronized (firstLock) {
 
+	public static void testCase1() throws InterruptedException {
+		final Object firstLock = new LockObject("firstLock");
+		final Object secondLock = new LockObject("secondLock");
+
+		synchronized (firstLock) {
 			Thread otherThread = new Thread() {
+				@Override
 				public void run() {
 					System.out.println("testCase1 :: Secondary thread trying to acquire lock on secondLock");
 					synchronized (secondLock) {
-						synchronized(mutex) {
+						synchronized (mutex) {
 							mutex.notify();
 						}
 						System.out.println("testCase1 :: Secondary thread trying to acquire lock on firstLock");
-						synchronized(firstLock) {
+						aboutToDeadlock();
+						synchronized (firstLock) {
 							System.out.println("Should not see me, testCase1()");
 						}
 					}
 				}
 			};
-			
-			synchronized(mutex) {
+
+			synchronized (mutex) {
 				otherThread.start();
 				mutex.wait();
 				System.out.println("testCase1 :: Main thread trying to acquire lock on secondLock");
+				aboutToDeadlock();
 				synchronized (secondLock) {
 					System.out.println("Should not see me, testCase1().");
 				}
 			}
 		}
 	}
-	
+
 	/* *****************************
 	 *         Test Case #2
-	 *	     JOS, JSO deadlock.
+	 *       JOS, JSO deadlock.
 	 * *****************************/
-	
-	public static void testCase2() throws InterruptedException 
-	{
+
+	public static void testCase2() throws InterruptedException {
 		setup();
-		
-		final Object javaObjMon = new Object();
-		
+
+		final Object javaObjMon = new LockObject("javaObjMon");
+
 		synchronized (javaObjMon) {
-			
 			Thread otherThread = new Thread() {
+				@Override
 				public void run() {
 					System.out.println("testCase2 :: Secondary thread trying to acquire lock on native monitor.");
-						enterFirstMonitor();
-						synchronized(mutex) {
-							mutex.notify();
-						}
-						System.out.println("testCase2 :: Secondary thread trying to acquire lock on Java object monitor.");
-						synchronized(javaObjMon) {
-							System.out.println("Should not see me, testCase2()");
-						}
+					enterFirstMonitor();
+					synchronized (mutex) {
+						mutex.notify();
+					}
+					System.out.println("testCase2 :: Secondary thread trying to acquire lock on Java object monitor.");
+					aboutToDeadlock();
+					synchronized (javaObjMon) {
+						System.out.println("Should not see me, testCase2()");
+					}
 				}
 			};
-			
-			synchronized(mutex) {
+
+			synchronized (mutex) {
 				otherThread.start();
 				mutex.wait();
 				System.out.println("testCase2 :: Main thread trying to acquire lock on native monitor.");
-				enterFirstMonitor(); // Deadlock
+				aboutToDeadlock();
+				enterFirstMonitor(); // deadlock
 				System.out.println("Should not see me, testCase2().");
-			}	
+			}
 		}
 	}
-	
+
 	/* *****************************
 	 *         Test Case #3
-	 *	     JSS, JSS deadlock.
+	 *       JSS, JSS deadlock.
 	 * *****************************/
-	
-	public static void testCase3() throws InterruptedException 
-	{
+
+	public static void testCase3() throws InterruptedException {
 		setup();
-		
+
 		enterFirstMonitor();
 
 		Thread otherThread = new Thread() {
+			@Override
 			public void run() {
 				System.out.println("testCase3 :: Secondary thread trying to acquire lock on secondLock");
-					enterSecondMonitor();
-					synchronized(mutex) {
-						mutex.notify();
-					}
-					System.out.println("testCase3 :: Secondary thread trying to acquire lock on firstLock");
-					enterFirstMonitor(); // Deadlock
-					System.out.println("Should not see me, testCase3()");
+				enterSecondMonitor();
+				synchronized (mutex) {
+					mutex.notify();
+				}
+				System.out.println("testCase3 :: Secondary thread trying to acquire lock on firstLock");
+				aboutToDeadlock();
+				enterFirstMonitor(); // deadlock
+				System.out.println("Should not see me, testCase3()");
 			}
 		};
 
-		synchronized(mutex) {	
+		synchronized (mutex) {
 			otherThread.start();
 			mutex.wait(); // Make sure the other thread has the monitor we want to block on first.
 			System.out.println("testCase3 :: Main thread trying to acquire lock on secondLock");
-			enterSecondMonitor(); // Deadlock
-			System.out.println("Should not see me, testCase3().");	
+			aboutToDeadlock();
+			enterSecondMonitor(); // deadlock
+			System.out.println("Should not see me, testCase3().");
 		}
 	}
-	
+
 	/* *****************************
 	 *         Test Case #4
-	 *	     JSS, NSS deadlock.
+	 *       JSS, NSS deadlock.
 	 * *****************************/
-	
-	public static void testCase4() 
-	{
+
+	public static void testCase4() {
 		setup();
 		enterFirstMonitor();
 		spawnNativeThread(); // Will return once second monitor has been acquired by the new native thread.
-		enterSecondMonitor(); // Deadlock.
+		aboutToDeadlock();
+		enterSecondMonitor(); // deadlock
 	}
-	
+
 	/* *****************************
 	 *         Test Case #5
-	 *	     NSS, NSS deadlock.
+	 *       NSS, NSS deadlock.
 	 * *****************************/
-	
-	public static void testCase5() 
-	{
+
+	public static void testCase5() {
 		setup();
 		createNativeDeadlock();
 	}
-	
+
 	/* *****************************
 	 *         Test Case #6
 	 * Test JOS, NSS, JSO deadlock.
 	 * *****************************/
-	
-	private static final Object lock1Test6 = new Object();
-	
-	public static void testCase6() 
-	{
+
+	private static final Object lock1Test6 = new LockObject("lock1Test6");
+
+	public static void testCase6() {
 		setup(); // Initializes native monitors.
-		
+
 		Thread thread1test6 = new Thread() {
+			@Override
 			public void run() {
 				synchronized (lock1Test6) { // Get O1
 					System.out.println("spawnFirstThreadtest6 :: run()");
 					spawnSecondThreadtest6();
 					try {
-						synchronized(mutex) {
+						synchronized (mutex) {
 							mutex.wait(); // Make sure second Java thread has the system monitor first!
 						}
 					} catch (InterruptedException e) {
@@ -267,25 +308,27 @@ public final class DeadlockCoreGenerator {
 					spawnNativeThread();
 					// At this point the native thread acquired S2 and wants S1.
 					System.out.println("spawnFirstThreadtest6 :: wait() indefinitely on enterSecondMonitor()");
-					enterSecondMonitor(); // Deadlock
+					aboutToDeadlock();
+					enterSecondMonitor(); // deadlock
 					System.out.println("Should not see me, spawnFirstThreadtest6().");
 				}
 			}
 		};
 		thread1test6.start();
 	}
-	
-	private static void spawnSecondThreadtest6()
-	{
+
+	private static void spawnSecondThreadtest6() {
 		Thread thread2test6 = new Thread() {
+			@Override
 			public void run() {
 				System.out.println("spawnSecondThreadtest6 :: run()");
 				enterFirstMonitor();
-				synchronized(mutex) {
+				synchronized (mutex) {
 					mutex.notify();
 				}
 				System.out.println("spawnSecondThreadtest6 :: block indefinitely on lock1test6");
-				synchronized(lock1Test6) { // Deadlock
+				aboutToDeadlock();
+				synchronized (lock1Test6) { // deadlock
 					System.out.println("Should not see me, spawnSecondThreadtest6().");
 				}
 			}

--- a/test/functional/DDR_Test/src/j9vm/test/ddrext/Constants.java
+++ b/test/functional/DDR_Test/src/j9vm/test/ddrext/Constants.java
@@ -633,8 +633,7 @@ public class Constants {
 	public static final String DEADLOCK_OWNED_BY = "which is owned by:";
 	public static final String DEADLOCK_FIRST_MON = "First Monitor lock";
 	public static final String DEADLOCK_SECOND_MON = "Second Monitor lock";
-	public static final String DEADLOCK_JAVA_OBJ = "java/lang/Object";
-	public static final String DEADLOCK_JAVA_IDENTITY = "java/lang/Identity";
+	public static final String DEADLOCK_JAVA_OBJ = "j9vm/test/corehelper/DeadlockCoreGenerator\\$LockObject";
 	public static final String DEADLOCK_CMD = "deadlock";
 
 	/* Constants related to testing of runtime type resolution */

--- a/test/functional/DDR_Test/src/j9vm/test/ddrext/DDRExtTesterBase.java
+++ b/test/functional/DDR_Test/src/j9vm/test/ddrext/DDRExtTesterBase.java
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2001, 2020 IBM Corp. and others
+ * Copyright (c) 2001, 2022 IBM Corp. and others
  *
  * This program and the accompanying materials are made available under
  * the terms of the Eclipse Public License 2.0 which accompanies this
@@ -40,35 +40,35 @@ import com.ibm.j9ddr.tools.ddrinteractive.CommandUtils;
 import com.ibm.j9ddr.tools.ddrinteractive.DDRInteractive;
 
 public class DDRExtTesterBase extends TestCase {
-	private static Logger log = Logger.getLogger(DDRExtTesterBase.class);
-	public static ArrayList<String> coveredStructures = new ArrayList<String>();
-	private String currentCommand = null;
-	private String[] coreinfoOutput = null;
+	private static final Logger log = Logger.getLogger(DDRExtTesterBase.class);
+	public static final ArrayList<String> coveredStructures = new ArrayList<>();
+
+	private String currentCommand;
+	private String[] coreinfoOutput;
 	private int javaVersion = -1;
 	private int javaSr = -1;
-	private String vmVersion = null;
+	private String vmVersion;
 
 	/**
-	 * Executes a given DDR extension and returns its output 
-	 * @param ddrExtCmd - DDR command to execute 
-	 * @param arg - Argument of the DDR command 
-	 * @return - Output of the DDR command 
+	 * Executes a given DDR extension and returns its output
+	 * @param ddrExtCmd - DDR command to execute
+	 * @param arg - Argument of the DDR command
+	 * @return - Output of the DDR command
 	 */
 	protected String exec(String ddrExtCmd, String[] arg) {
+		String argStr = "";
+		for (int i = 0; i < arg.length; i++) {
+			argStr = argStr + " " + arg[i];
+		}
 		String output = null;
-		if (SetupConfig.getDDRContxt() != null) { // this means we are
-			// running from ddr
-			// plugin
+		if (SetupConfig.getDDRContxt() != null) {
+			// this means we are running from ddr plugin
 			DDROutputStream mps = SetupConfig.getPrintStream();
-			String argStr = "";
-			for (int i = 0; i < arg.length; i++) {
-				argStr = argStr + " " + arg[i];
-			}
-			log.debug("Execute command: !" + ddrExtCmd + " " + argStr);
+			log.debug("Execute command: !" + ddrExtCmd + argStr);
 			if (ddrExtCmd.startsWith("!") == false) {
 				ddrExtCmd = "!" + ddrExtCmd;
 			}
-			currentCommand = "'" + ddrExtCmd + " " + argStr + "'";
+			currentCommand = "'" + ddrExtCmd + argStr + "'";
 			SetupConfig.getDDRContxt().execute(ddrExtCmd, arg,
 					SetupConfig.getPrintStream());
 			output = mps.getOutBuffer().toString();
@@ -89,15 +89,11 @@ public class DDRExtTesterBase extends TestCase {
 			}
 			if (app != null) {
 				DDROutputStream mps = SetupConfig.getPrintStream();
-				String argStr = "";
-				for (int i = 0; i < arg.length; i++) {
-					argStr = argStr + " " + arg[i];
-				}
-				log.debug("Execute command: !" + ddrExtCmd + " " + argStr);
+				log.debug("Execute command: !" + ddrExtCmd + argStr);
 				if (ddrExtCmd.startsWith("!") == false) {
 					ddrExtCmd = "!" + ddrExtCmd;
 				}
-				currentCommand = "'" + ddrExtCmd + " " + argStr + "'";
+				currentCommand = "'" + ddrExtCmd + argStr + "'";
 				app.execute(ddrExtCmd, arg);
 				output = mps.getOutBuffer().toString();
 				mps.clear();
@@ -117,8 +113,11 @@ public class DDRExtTesterBase extends TestCase {
 	 * @return - True if validation passes,false otherwise
 	 */
 	private boolean basicValidation(String output) {
-		if (output == null || output.isEmpty()) {
-			log.error(currentCommand + " output is null or empty");
+		if (output == null) {
+			log.error(currentCommand + " output is null");
+			return false;
+		} else if (output.isEmpty()) {
+			log.error(currentCommand + " output is empty");
 			return false;
 		} else if (output.contains(Constants.UNRECOGNIZED_CMD)) {
 			log.error(currentCommand + " is not recognized");
@@ -126,38 +125,35 @@ public class DDRExtTesterBase extends TestCase {
 		}
 		return true;
 	}
-	
+
 	/**
 	 * Validation method for DDR extension output. Check that successKey appears expectedMatches number of times
-	 * 
+	 *
 	 * @param output - DDR extension output to validate.
 	 * @param successKey - Comma separated list of tokens that must be available in the output for it to be valid.
 	 * @param expectedMatches - number of times successKey must be seen
-	 * @return True if validation passes, false otherwise. 
+	 * @return True if validation passes, false otherwise.
 	 */
 	protected boolean validate(String output, String successKey, int expectedMatches) {
-		
-		int matchesFound = 0;
-		
 		Pattern pattern = Pattern.compile(successKey.trim(), Pattern.CASE_INSENSITIVE);
 		Matcher matcher = pattern.matcher(output);
-		
+		int matchesFound = 0;
+
 		while (matcher.find()) {
 			matchesFound = matchesFound + 1;
 		}
-		
-		return (matchesFound == expectedMatches);
+
+		return matchesFound == expectedMatches;
 	}
-	
+
 	/**
 	 * Validation method for DDR extension output
 	 * @param output - DDR extension output to validate.
 	 * @param successKeys - Comma separated list of tokens that must be available in the output for it to be valid.
 	 * @param failureKeys - Comma separated list of tokens that must NOT be available in the output for it to be valid.
-	 * @return True if validation passes, false otherwise. 
+	 * @return True if validation passes, false otherwise.
 	 */
-	protected boolean validate(String output, String successKeys,
-			String failureKeys) {
+	protected boolean validate(String output, String successKeys, String failureKeys) {
 		return validate(output, successKeys, failureKeys, false, false);
 	}
 
@@ -167,7 +163,7 @@ public class DDRExtTesterBase extends TestCase {
 	 * @param successKeys - Comma separated list of tokens that must be available in the output for it to be valid.
 	 * @param failureKeys - Comma separated list of tokens that must NOT be available in the output for it to be valid.
 	 * @param goNextStep - Flag indicating whether or not we will perform exhaustive validation using all extension outputs available in this output.
-	 * @return True if validation passes, false otherwise. 
+	 * @return True if validation passes, false otherwise.
 	 */
 	protected boolean validate(String output, String successKeys,
 			String failureKeys, boolean goNextStep) {
@@ -181,7 +177,7 @@ public class DDRExtTesterBase extends TestCase {
 	 * @param failureKeys - Comma separated list of tokens that must NOT be available in the output for it to be valid.
 	 * @param goNextStep - Flag indicating whether or not we will perform exhaustive validation using all extension outputs available in this output.
 	 * @param isNegativeTest - Flag indicating whether or not this is a negative test. If set, it will not fail validation if it sees a failure key in output.
-	 * @return True if validation passes, false otherwise. 
+	 * @return True if validation passes, false otherwise.
 	 */
 	protected boolean validate(String output, String successKeys,
 			String failureKeys, boolean goNextStep, boolean isNegativeTest) {
@@ -229,13 +225,13 @@ public class DDRExtTesterBase extends TestCase {
 			}
 		}
 
-		/*Performing exhaustive validation using all available structures (subcommands) in the output*/
+		/* Performing exhaustive validation using all available structures (subcommands) in the output */
 		if (goNextStep) {
 			int firstIndex = output.indexOf("!");
 			String parentCmd = currentCommand;
 			boolean allNextCmdAlreadyCovered = true;
 			if (firstIndex != -1) { // if this output contains any !subcommand
-				ArrayList<String> structuresToRun = new ArrayList<String>();
+				ArrayList<String> structuresToRun = new ArrayList<>();
 				output = output.substring(output.indexOf("!") + 1); // chop off
 				Scanner scanner = new Scanner(output).useDelimiter("!");
 				String structToRun = null;
@@ -280,11 +276,9 @@ public class DDRExtTesterBase extends TestCase {
 
 						if (nxtOutput.contains("{")
 								&& nxtOutput.contains("}")
-								&& nxtOutput.indexOf("{") < nxtOutput
-								.indexOf("}")) {
+								&& nxtOutput.indexOf("{") < nxtOutput.indexOf("}")) {
 							String structPropertyPattern = ".*=.* 0x.*|<FAULT>";
-							Pattern pattern = Pattern
-							.compile(structPropertyPattern);
+							Pattern pattern = Pattern.compile(structPropertyPattern);
 							Matcher matcher = pattern.matcher(nxtOutput);
 							if (matcher.find()
 									&& runCommonFailureTestForSubCmd(nxtOutput)) {
@@ -314,7 +308,7 @@ public class DDRExtTesterBase extends TestCase {
 	 * @param address
 	 * @return
 	 */
-	private String cleanse(String address) {
+	private static String cleanse(String address) {
 		if (address.contains(Constants.NL)) {
 			address = address.substring(0, address.indexOf(Constants.NL));
 		}
@@ -331,7 +325,7 @@ public class DDRExtTesterBase extends TestCase {
 		if (address.contains("\t")) {
 			address = address.split("\t")[0];
 		}
-	
+
 		if (address.endsWith(")") || address.endsWith(">")) {
 			address = address.substring(0, address.length() - 1);
 		}
@@ -346,8 +340,8 @@ public class DDRExtTesterBase extends TestCase {
 	private boolean runCommonFailureTest(String output) {
 		String[] cFailKeys = Constants.COMMON_FAILURE_KEYS.split(",");
 		for (String cfKey : cFailKeys) {
-			Pattern pattern = Pattern.compile(cfKey, Pattern.CASE_INSENSITIVE
-					| Pattern.LITERAL);
+			Pattern pattern = Pattern.compile(cfKey,
+					Pattern.CASE_INSENSITIVE | Pattern.LITERAL);
 			Matcher matcher = pattern.matcher(output);
 			if (matcher.find()) {
 				log.error(currentCommand
@@ -360,7 +354,7 @@ public class DDRExtTesterBase extends TestCase {
 	}
 
 	/**
-	 * Common failure test for subcommands. Used during exhaustive validation of a ddr extension. 
+	 * Common failure test for subcommands. Used during exhaustive validation of a ddr extension.
 	 * @param output
 	 * @return
 	 */
@@ -371,8 +365,7 @@ public class DDRExtTesterBase extends TestCase {
 				if (cfKey.trim().equals("<FAULT>")) {
 					log.warn(currentCommand
 							+ " output contains <FAULT> references");
-					// log.warn(currentCommand+ " output :" + Constants.NL +
-					// output);
+					// log.warn(currentCommand+ " output :" + Constants.NL + output);
 					return true;
 				}
 				log.error(currentCommand
@@ -384,35 +377,34 @@ public class DDRExtTesterBase extends TestCase {
 		return true;
 	}
 
-	
 	/**
 	 * Get the address corresponding to the cmd in threadOutput
-	 * 
+	 *
 	 * For example, if we wanted the address 0x03596100 corresponding to j9vmthread in the following string:
-	 * 
+	 *
 	 * !stack 0x03596100	!j9vmthread 0x03596100	!j9thread 0x2aaab7c72b10	tid 0x5221 (21025) // (Thread-3)
-	 * 
+	 *
 	 * 	we would call getAddressFor with:
-	 * 
+	 *
 	 *  	threadOutput = the string
 	 *  	eyeCatcher 	= "(Thread-3"
 	 *  	cmd  		= "j9vmthread"
 	 *  	cmdDelimiter= "\t"
-	 *  
+	 *
 	 * @param threadOutput	the output to parses
 	 * @param eyeCatcher	only search this string if eyeCatcher is present
 	 * @param cmd			the cmd who's address we are after
 	 * @param cmdDelimiter	the delimiter that separates multiples commands in the output
-	 * @return 	the string representation of the address as it appears in threadOutput 
+	 * @return 	the string representation of the address as it appears in threadOutput
 	 */
 	public String getAddressFor(String threadOutput, String eyeCatcher, String cmd, String cmdDelimiter) {
 		String[] lines = threadOutput.split(Constants.NL);
-		for (int i = 0; i < lines.length; i++) {
-			if ((null == eyeCatcher) || (lines[i].contains(eyeCatcher))) {
-				String[] tokens = lines[i].split(cmdDelimiter);
-				for (int j = 0; j < tokens.length; j++) {
-					if (tokens[j].contains(cmd)) {
-						return tokens[j].trim().split(" ")[1];
+		for (String line : lines) {
+			if ((null == eyeCatcher) || line.contains(eyeCatcher)) {
+				String[] tokens = line.split(cmdDelimiter);
+				for (String token : tokens) {
+					if (token.contains(cmd)) {
+						return token.trim().split(" ")[1];
 					}
 				}
 			}
@@ -420,12 +412,11 @@ public class DDRExtTesterBase extends TestCase {
 		return null;
 	}
 
-
 	/**
 	 * Helper function that looks for the address cmd in threadOutput
-	 * 
+	 *
 	 * See {@link DDRExtTesterBase#getAddressFor(String, String, String, String)}
-	 * 
+	 *
 	 * @param cmd
 	 * @param threadOutput
 	 * @return the address corresponding to the command
@@ -433,7 +424,7 @@ public class DDRExtTesterBase extends TestCase {
 	public String getAddressForThreads(String cmd, String threadOutput) {
 		return getAddressFor(threadOutput, "(Thread-", cmd, "\t");
 	}
-	
+
 	/**Used to get the full core file path from the ddrext.properties file
 	 * @return
 	 */
@@ -458,7 +449,7 @@ public class DDRExtTesterBase extends TestCase {
 			log.error("Not able to read file " + Constants.DDREXT_PROPERTIES);
 			log.error(e.toString());
 		}
-		if (path.equals("")) {
+		if ((path == null) || path.isEmpty()) {
 			log.error("Please provide your dump location in "
 					+ Constants.DDREXT_PROPERTIES);
 			return null;
@@ -469,20 +460,20 @@ public class DDRExtTesterBase extends TestCase {
 	public static void resetList() {
 		coveredStructures.clear();
 	}
-	
-	/* Check whether core is generated on 64 bit platform or not. 
-	 * Run setvm with the address that can exist only on 64 bit platform 
+
+	/* Check whether core is generated on 64 bit platform or not.
+	 * Run setvm with the address that can exist only on 64 bit platform
 	 */
-	public boolean is64BitPlatform(){
-		String setVMOutput = exec(Constants.SETVM_CMD, new String[] {CommandUtils.UDATA_MAX_64BIT.toString()});
+	public boolean is64BitPlatform() {
+		String setVMOutput = exec(Constants.SETVM_CMD, new String[] { CommandUtils.UDATA_MAX_64BIT.toString() });
 		boolean is64BitPlatform = true;
-		/* If it is 32 bit platform, than it will complain about address being too large */
+		/* If it is 32-bit platform, then it will complain about address being too large. */
 		if (validate(setVMOutput, "is larger than the max available memory address: 0xFFFFFFFF", null)) {
 			is64BitPlatform = false;
-		} 
+		}
 		return is64BitPlatform;
 	}
-	
+
 	/**
 	 * get and save the output from !coreinfo
 	 * @return array of strings
@@ -555,7 +546,7 @@ public class DDRExtTesterBase extends TestCase {
 		javaVersion = 0;
 		return javaVersion;
 	}
-	
+
 	/**
 	 * @return the version string for the J9 VM, e.g. "2.6", or "unknown"
 	 */

--- a/test/functional/DDR_Test/src/j9vm/test/ddrext/junit/deadlock/TestDeadlockCase1.java
+++ b/test/functional/DDR_Test/src/j9vm/test/ddrext/junit/deadlock/TestDeadlockCase1.java
@@ -29,22 +29,22 @@ public class TestDeadlockCase1 extends DDRExtTesterBase
 	/**
 	 * Test !monitors deadlock
 	 *      Test Case #1
-	 *	  JOO, JOO deadlock.
+	 *    JOO, JOO deadlock.
 	 */
-	
+
 	public void testDeadlock1()
 	{
 		String output = exec(Constants.MONITORS_CMD, new String[] { Constants.DEADLOCK_CMD });
-		
+
 		if (null == output) {
 			fail("\"!monitors deadlock\" output is null. Can not proceed with test");
 			return;
 		}
-		
+
 		assertTrue(validate(output, Constants.DEADLOCK_THREAD, 3));
 		assertTrue(validate(output, Constants.DEADLOCK_BLOCKING_ON, 2));
 		assertTrue(validate(output, Constants.DEADLOCK_OWNED_BY, 2));
-		assertTrue(validate(output, Constants.DEADLOCK_JAVA_OBJ, 2) || validate(output, Constants.DEADLOCK_JAVA_IDENTITY, 2));
+		assertTrue(validate(output, Constants.DEADLOCK_JAVA_OBJ, 2));
 	}
 
 }

--- a/test/functional/DDR_Test/src/j9vm/test/ddrext/junit/deadlock/TestDeadlockCase2.java
+++ b/test/functional/DDR_Test/src/j9vm/test/ddrext/junit/deadlock/TestDeadlockCase2.java
@@ -29,22 +29,22 @@ public class TestDeadlockCase2 extends DDRExtTesterBase
 	/**
 	 * Test !monitors deadlock
 	 *     Test Case #2
-	 *	 JOS, JSO deadlock.
+	 *   JOS, JSO deadlock.
 	 */
-	
+
 	public void testDeadlock2()
 	{
 		String output = exec(Constants.MONITORS_CMD, new String[] { Constants.DEADLOCK_CMD });
-		
+
 		if (null == output) {
 			fail("\"!monitors deadlock\" output is null. Can not proceed with test");
 			return;
 		}
-		
+
 		assertTrue(validate(output, Constants.DEADLOCK_THREAD, 3));
 		assertTrue(validate(output, Constants.DEADLOCK_BLOCKING_ON, 2));
 		assertTrue(validate(output, Constants.DEADLOCK_OWNED_BY, 2));
-		assertTrue(validate(output, Constants.DEADLOCK_JAVA_OBJ, 1) || validate(output, Constants.DEADLOCK_JAVA_IDENTITY, 1));
+		assertTrue(validate(output, Constants.DEADLOCK_JAVA_OBJ, 1));
 		assertTrue(validate(output, Constants.DEADLOCK_FIRST_MON, 1));
 	}
 

--- a/test/functional/DDR_Test/src/j9vm/test/ddrext/junit/deadlock/TestDeadlockCase6.java
+++ b/test/functional/DDR_Test/src/j9vm/test/ddrext/junit/deadlock/TestDeadlockCase6.java
@@ -31,26 +31,26 @@ public class TestDeadlockCase6 extends DDRExtTesterBase
 	 *        Test Case #6
 	 * Test JOS, NSS, JSO deadlock.
 	 */
-	
+
 	public void testDeadlock6()
 	{
 		String output = exec(Constants.MONITORS_CMD, new String[] { Constants.DEADLOCK_CMD });
-		
+
 		if (null == output) {
 			fail("\"!monitors deadlock\" output is null. Can not proceed with test");
 			return;
 		}
-		
+
 		// N.B. The order in which the threads are found is not well defined,
 		// thus in this case we may have "OS Thread" or "Thread" appear twice.
-		
+
 		assertTrue(validate(output, Constants.DEADLOCK_THREAD, null));
 		assertTrue(validate(output, Constants.DEADLOCK_OS_THREAD, null));
 		assertTrue(validate(output, Constants.DEADLOCK_BLOCKING_ON, 3));
 		assertTrue(validate(output, Constants.DEADLOCK_OWNED_BY, 3));
 		assertTrue(validate(output, Constants.DEADLOCK_FIRST_MON, 1));
 		assertTrue(validate(output, Constants.DEADLOCK_SECOND_MON, 1));
-		assertTrue(validate(output, Constants.DEADLOCK_JAVA_OBJ, 1) || validate(output, Constants.DEADLOCK_JAVA_IDENTITY, 1));
+		assertTrue(validate(output, Constants.DEADLOCK_JAVA_OBJ, 1));
 	}
 
 }

